scripts/speech_recognition/code_switching/code_switching_audio_data_creation.py

```diff
@@ -0,0 +1,283 @@
+            staring_pause = np.zeros(int(pause_beg_msec * fs / 1000))
+            combined_audio += list(staring_pause)
+
+            for index in range(len(data['lang_ids'])):
+
+                data_sample, fs_sample = librosa.load(data['paths'][index], sr=fs)
+                # Alternative-  fs_sample, data_sample = wavfile.read(data['paths'][index])
+
+                if fs_sample != fs:
+                    logging.error('Sampling rate error inside create_cs_data function')
+                    exit
+
+                # Remove leading and trailing zeros
+                data_sample = np.trim_zeros(data_sample)
+
+                # take care of empty arrays: rare
+                if data_sample.size == 0:
+                    incorrect_sample_flag = 1
+                    continue
+
+                # normalizing data
+                data_sample_norm = (
+                    data_sample
+                    / np.maximum(np.abs(data_sample.max()), np.abs(data_sample.min()))
+                    * audio_amplitude_normalization
+                )
+
+                combined_audio += list(data_sample_norm)
+
+                # adding small pause between gemgments
+                if index != (len(data['lang_ids']) - 1):
+                    pause = np.zeros(int(pause_join_msec * fs / 1000))
+                    combined_audio += list(pause)
+
+            if incorrect_sample_flag == 1:
+                incorrect_sample_flag = 0
+                continue
+
+            ending_pause = np.zeros(int(pause_end_msec * fs / 1000))
+            combined_audio += list(ending_pause)
+
+            sample_id = data['uid']
+            audio_file_path = audio_save_folder + '/' + str(sample_id) + ".wav"
+
+            # saving audio
+            wavfile.write(audio_file_path, fs, np.array(combined_audio).astype(np.int16))
+            # Alternative-  librosa.output.write_wav(audio_file_path, combined_audio, fs)
+
+            metadata_json = {}
+            metadata_json['audio_filepath'] = audio_file_path
+            metadata_json['duration'] = float(len(combined_audio) / fs)
+            metadata_json['text'] = ' '.join(data['texts'])
+
+            metadata_json['language_ids'] = data['lang_ids']
+            metadata_json['original_texts'] = data['texts']
+            metadata_json['original_paths'] = data['paths']
+            metadata_json['original_durations'] = data['durations']
+
+            s = json.dumps(metadata_json)
+            outfile.write(s + '\n')
+
+
+def main():
+
+    cs_intermediate_manifest_path = args.manifest_path
+    audio_save_folder = args.audio_save_folder_path
+    manifest_save_path = args.manifest_save_path
+    audio_amplitude_normalization = args.audio_normalized_amplitude
+    pause_beg_msec = args.sample_beginning_pause_msec
+    pause_join_msec = args.sample_joining_pause_msec
+    pause_end_msec = args.sample_end_pause_msec
+    cs_data_sampling_rate = args.cs_data_sampling_rate
+    num_process = args.num_process
```

Comment:
Was this tested? Other script call this parameter "workers" -- e,g, https://github.com/NVIDIA/NeMo/blob/c74bfc093fe151bfb275c1fc1aa266832f7036ea/scripts/speech_recognition/convert_to_tarred_audio_dataset.py#L177 ; we need to pass it in and document.

Reply:
Yes, this was tested locally. Changed the parameter name